src/MyMap/MyHashMap.java

```diff
@@ -48,25 +57,26 @@ public void put(Object key, Object value) {
 
             } else if (capacity - size == 1) {
                 //TODO change name to something like normalizeSize
-                resize();
+                normalizeMap();
             }
 
+
             Object[] pairKeyValue = new Object[2];
             pairKeyValue[0] = key;
             pairKeyValue[1] = value;
 
+
+
             size++;
 
             // can we have version with duplicate keys?
             // yes, maybe the user wants to put an element with the key which is already exists.
 
-            if (checkUniqueKey(key, value)) {
+            if (GetRemoveHelper(key, Flag.GET) == null) {
                 arr[getHashCode(key)].add(pairKeyValue);
-                /**
-                 if (arr[getHashCode(key)].size() == 3) {
-                 newSpace();
-                 }
-                 */
+//                arr[getHashCode(key)].add(new Node (key, value));
+            }else {
+                GetRemoveHelper(key, Flag.GET)[1] = value;
```

Comment:
can you store result of GetRemoveHelper(key, Flag.GET) for not call it again in else statement?

Reply:
yes, of course